src/services/services.ts

```diff
@@ -488,6 +502,63 @@ namespace ts {
         }
     }
 
+    /**
+     * Returns whether or not the given symbol or signature has a JSDoc "inheritDoc" tag on it.
+     * @param symbol the Symbol or Signature in question.
+     * @returns `true` if `symbol` has a JSDoc "inheritDoc" tag on it, otherwise `false`.
+     */
+    function hasJSDocInheritDocTag(symbol: Signature | Symbol) {
+        return !!find(symbol.getJsDocTags(), tag => tag.name === "inheritDoc");
+    }
+
+    /**
+     * Attempts to find JSDoc comments for possibly-inherited properties.  Checks superclasses then traverses
+     * implemented interfaces until a symbol is found with the same name and with documentation.
+     * @param declaration The possibly-inherited declaration to find comments for.
+     * @param propertyName The name of the possibly-inherited property.
+     * @param typeChecker A TypeChecker, used to find inherited properties.
+     * @returns A filled array of documentation comments if any were found, otherwise an empty array.
+     */
+    function findInheritedJSDocComments(declaration: Declaration, propertyName: string, typeChecker: TypeChecker): SymbolDisplayPart[] {
+        let documentationComment: SymbolDisplayPart[] = [];
+
+        if (isClassDeclaration(declaration.parent) || isInterfaceDeclaration(declaration.parent)) {
+            const container: ClassDeclaration | InterfaceDeclaration = declaration.parent;
+            const baseTypeNode = getClassExtendsHeritageClauseElement(container);

```

Comment:
There's a function `getBaseTypes` in `checker.ts` that might be better suited. You could expose it internally by adding `/* @internal */ getBaseTypes(): ReadonlyArray<BaseType>;` to `interface TypeChecker`.
The advantages are that this gets you both classes and interfaces, and you don't have to call `getTypeAtLocation`.